src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -520,7 +538,60 @@ private void checkWhetherMetadataExistsBeforeSave(SettingConfiguration entity, L
 				settingsMetadataList.add(metadata);
 			}
 		}
-			settingMetadataMapper.insertMany(settingsMetadataList);
+		insertSettingMetadata(settingsMetadataList);
+	}
+
+	private void insertSettingMetadata(List<SettingsMetadata> settingsMetadataList) throws SQLException {
```

Comment:
If errors do happen and some records are not saved should the identifiers be sent back to the API especially on this catch block
```
catch (JsonProcessingException e)
```